rdb/src/main/java/org/openmbee/mms/rdb/repositories/BaseDAOImpl.java

```diff
@@ -36,6 +38,10 @@ public JdbcTemplate getConn() {
 
     public String getSuffix() {
         String refId = ContextHolder.getContext().getBranchId();
-        return refId.equals(ContextObject.MASTER_BRANCH) ? "" : refId.toLowerCase();
+        if(Pattern.matches("^[a-zA-Z0-9-_]*$", refId)) { 
```

Comment:
getSuffix() is used in a number of places multiple times in our DAOs. Recompiling the pattern every time could potentially have an affect on performance, albeit, probably not significantly.

Reply:
there's a BRANCH_ID_VALID_PATTERN in the BranchesController - may be better if that gets moved to core Formats?

Reply:
i think so

Reply:
updated to use a common Pattern